### PR TITLE
[TCES-27]: GET Request for Job Applications

### PR DESCRIPTION
**Pull request by williamntlam**
# Description

An implementation of getting Job Applications via GET requests. There are two functionalities: Getting all job applications and getting all job applications with a  specific job posting id.

Testing getting all Job Applications
image

Testing getting all Job Applications with a specific job posting id.
image

Testing getting all Job Applications with a specific job posting id when the id doesn't exist.
image

TODO: Testing getting all Job Applications with a specific job posting id where there are multiple job applications.

Issue #TCES-27

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

Please delete options that are not applicable.

- [X] My code follows the style guidelines of this project (required)
- [X] I have performed a self-review of my code (required)
- [X] I have linted and formatted my code (required)
- [X] I have added tests that prove my fix is effective or that my feature works

Testing:

Uncomment lines 89-143 to create objects in the database to satisfy the foreign key constraints for the Job Application posting.
    run index.js
    Use the Postman test below to test the POST request for creating a Job Application:

    URL: http://localhost:8000/job_applications
    JSON Body:

{
"job_posting_id": 1,
"name": "John Doe",
"email": "[johndoe@example.com](mailto:johndoe@example.com)",
"phone": "1234567890",
"postal_code": "A1A 1A1",
"resume": "resume_file.pdf",
"status_in_canada": "Citizen",
"application_status": "New",
"custom_responses": {
"question1": "Yes",
"question2": "No"
}
}

Then, run the following Postman test cases:

http://localhost:8000/job_applications (Get request for all general job applications)
http://localhost:8000/job_applications?job_posting_id=1 (Get request for all job applications for job posting id = 1)
http://localhost:8000/job_applications?job_posting_id=1 (Get request for all job applications for job posting id = 2)

Hope that helps,
William.

### CODE REVIEW

**Comment by williamntlam** [comment]
Areas for Improvement

Admin Validation Middleware:

Right now, there's no middleware to check if a user has admin rights for searching Job Applications. I feel like this would be necessary and I'm going to add it soon.

William.

**Comment by 04kash** [comment]
> Right now, there's no middleware to check if a user has admin rights for searching Job Applications. I feel like this would be necessary and I'm going to add it soon.

We can have a separate ticket to make sure that all the endpoints we've added so far check for admin rights

**Comment by williamntlam** [comment]
Post Meeting Comment (Sunday November 10th 9pm): 

I still have to do the last task in order to be done with this PR. 

[X] Connect the frontend to your API endpoint by making a call to it in client/source/utils/api

[X] Implement functionality of getting the PDF from the S3 bucket with the given key.
The presignedURL is attached to the correct user now.

**Comment by williamntlam** [comment]
Just added the API calls for the frontend in api.js under the utils directory. I think this means that the PR is finished. Feel free to check it over and see if I made any mistakes!

William.

File: .prettierrc
Comment (04kash) [inline .prettierrc:1]: Just wondering why this was added?

File: server/src/configs/sequelize.js [hunk 1]
```diff
@@ -23,8 +26,31 @@ const sequelize = new Sequelize(
   },
 );
 
+// Authenticate the connection
 sequelize.authenticate();
 
-module.exports = {
-  sequelize,
-};
+// // Function to reinitialize database
```
Comment (04kash) [inline server/src/configs/sequelize.js]: Can this commented out code be removed?

File: server/src/controllers/job_applications/getAllJobApplications.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+const Sequelize = require("sequelize");
+const logger = require("pino")();
+const JobApplications = require("../../models/job_applications.model");
+
+const getAllJobApplicationsRequestHandler = async (req, res) => {
+  try {
+    // ! Return all applications sorted in descending order by application date (newest first)
+    // ! There is probably a sequelize function that can do this for me.
+    const query = {};
+    const order = [["applied_date", "DESC"]];
```
Comment (04kash) [inline server/src/controllers/job_applications/getAllJobApplications.js]: This PR removes the applied_date field from the model: https://github.com/uoftblueprint/tces/pull/127, please use the createdAt field instead

**Comment by williamntlam** [comment]
I think I resolved all the issues with the last review and the PR is ready for review again!

**Review by 04kash — APPROVED** [review]

**Pull request merged**